### PR TITLE
[IT-937] Add environment variables

### PR DESCRIPTION
**Pull request by zaro0508**
Setup lambda cloudformation template to allow user to
specify Synapse ID and bucket folders to sync and pass
those env vars to the lambda.

### CODE REVIEW

File: s3_synapse_sync/lambda_function.py [hunk 1]
```diff
@@ -22,23 +22,22 @@ def lambda_handler(event, context):
...
     ssm_user = '/HTAN/SynapseSync/username'
     ssm_api = '/HTAN/SynapseSync/apiKey'
-    ev_project = bucket+'_synapseProjectId'
-    ev_folders = bucket+'_foldersToSync'
 
     username = ssm.get_parameter(Name=ssm_user, WithDecryption=True)['Parameter']['Value']
     apiKey = ssm.get_parameter(Name=ssm_api, WithDecryption=True)['Parameter']['Value']
-    project_id = os.environ.get(ev_project, 'synapseProjectId variable is not set.')
-    inclFolders = os.environ.get(ev_folders, 'foldersToSync environment variable is not set.')
+    project_id = os.environ.get('SYNAPSE_PROJECT_ID', 'SYNAPSE_PROJECT_ID variable is not set.')
```
Comment (clarisse-lau) [inline s3_synapse_sync/lambda_function.py:28]: My understanding from the beginning of this project has been that each center will have a bucket to upload their data, and that data will be 'synced' to the center's specific Synapse project (not all to the same project). Thus, the lambda needs to know based on which bucket triggered it, which Synapse project the triggering object should be synced to.

**Comment by clarisse-lau** [comment]
@zaro0508 I haven't been able to think of any way to really get around the circular dependency... Can the environment variables be left out of the cloudformation template and just added to the deployed lambda via cli as needed?

File: README.md [hunk 2]
```diff
@@ -235,7 +235,9 @@ aws ssm put-parameter --name /HTAN/SynapseSync/<parameter> --value <value> --typ
 | `<bucket_name>_foldersToSync` | Comma separated list of folders in bucket to be synchronized to Synapse |
 
 ```
-aws lambda update-function-configuration --function-name <value> --environment Variables="{<bucket_name>_synapseProjectId=<value>,<bucket_name>_foldersToSync=<value>}"
+aws lambda update-function-configuration \
+--function-name <value> \
+--environment Variables="{SynapseProjectId=<value>, SynapseFoldersToSync=BucketA/FolderX,BucketA/FolderY,BucketB/FolderX}"
```
Comment (zaro0508) [inline README.md:240]: ideas @clarisse-lau 

How about pass in SynapseProjectIds and  SynapseFoldersToSync as lists, both containing the bucket name as a prefix? for example...
```
SynapseProjectIds
  - bucket-a::syn11111
  - bucket-b::syn2222
SynapseFoldersToSync
  - bucket-a::folderA
  - bucket-a::folderB
  - bucket-b::folderA
```

How about pass in a yaml file containing the dictionary of mappings?

```
bucket-a:
  SynapseProjectId: syn11111
  SynapseFoldersToSync:
    - folderA
    - folderB
```

Comment (clarisse-lau) [reply README.md:240]: Thanks for brainstorming this, @zaro0508 . In the first option, are these lists defined in the cloudformation template? I’m not opposed to making a switch from environment variables, but just want to better understand whether this change will take care of all permissions issues. 

For example, again looking at step 6 in https://github.com/Sage-Bionetworks/s3-synapse-sync/blob/master/README.md - this requires making a change to the lambda, so won’t a user need to have permissions to do this? And if they have that permission, they should be able to add environment variables?

Comment (zaro0508) [reply README.md:240]: > Thanks for brainstorming this, @zaro0508 . In the first option, are these lists defined in the cloudformation template? I’m not opposed to making a switch from environment variables, but just want to better understand whether this change will take care of all permissions issues.

If we let the service role in our CI/CD system update changes then there are no permissions issues.  Permissions issue only occur if we want to allow users access.

> For example, again looking at step 6 in https://github.com/Sage-Bionetworks/s3-synapse-sync/blob/master/README.md - this requires making a change to the lambda, so won’t a user need to have permissions to do this? And if they have that permission, they should be able to add environment variables?

I've address this in PR https://github.com/Sage-Bionetworks/scicomp-provisioner/pull/374

Comment (clarisse-lau) [reply README.md:240]: Ok, think I’m beginning to understand. I’ve been adding env vars to deployed lambdas just via cli (`aws lambda update-function-configuration --function-name <value> --environment Variables="{<bucket_name>_synapseProjectId=<value>,<bucket_name>_foldersToSync=<value>}”`)
But if we’d prefer the lambda to be configured fully in the template, I see that additions would be cumbersome bc of re-deployment each time. I will try to implement your yaml file mapping suggestion

**Comment by zaro0508** [comment]
This has been replaced with PR #7

**Pull request closed without merging**